File: Java/Data Structure/003. Selection Sort.java
import java.util.Scanner;
class SelectionSort{
    void Selection(int Arr[]){
        int i,j,Min,temp;
        for(i=0; i<Arr.length-1; i++){
            Min =i;                     // Let First has minimum value of the array
            for(j=i+1; j<Arr.length; j++){
                if(Arr[Min]>Arr[j])
                    Min =j;             // If found minimum value more thah that(first Min) change location to that.
            }
            // Swap minimum value with present index address.
            temp = Arr[i];
            Arr[i] = Arr[Min];
            Arr[Min] = temp;
        }   
    }
    void PrintArray(int Arr[]){
        for(int i=0; i<Arr.length; i++)
            System.out.print(Arr[i]+"  ");
        System.out.println();
    }
    public static void main(String[] args) {
        int n,i;
        Scanner Sc = new Scanner(System.in);
        SelectionSort obj = new SelectionSort();
        System.out.println("Enter Size of Array:");
        n = Sc.nextInt();
        int Arr[] = new int[n];
        System.out.println("Enter "+n+" Number");
        for(i=0; i<n; i++)
            Arr[i] = Sc.nextInt();
        System.out.println("Befor Sort:");
        obj.PrintArray(Arr);
        obj.Selection(Arr);
        System.out.println("After Selection Sort:");
        obj.PrintArray(Arr);
    }
}

File: C/Data Structure/003. Selection Sort.c
#include<stdio.h>
void PrintArray(int Arr[],int size){
    for(int i=0;i<size; i++){
        printf("%d\t",Arr[i]);
    }printf("\n");
}
int main(){
    int i,j,n,min,temp;
    printf("Enter Size of Array: ");
    scanf("%d",&n);
    int arr[n];
    printf("Enter %d Number:\n",n);
    for(i=0; i<n; i++){
        scanf("%d",&arr[i]);
    }
    printf("Before sort:\n");
    PrintArray(arr,n);
    for(i=0; i<n-1; i++){
        min = i;                    // Let First has minimum value of the array
        for(j=i+1; j<n; j++){
            if(arr[j]<arr[min]){
                min =j;             // If found minimum value more6 thah that(first Min) change location to that.
            }
        }
        // Swap minimum value with present index address.
        temp = arr[min];
        arr[min] = arr[i];
        arr[i] = temp;
    }
    printf("After Selection Sort:\n");
    PrintArray(arr,n);
}
